src/main/scala/com/scylladb/migrator/writers/DynamoDB.scala

```diff
@@ -23,27 +24,39 @@ object DynamoDB {
       if (readThroughput.isEmpty || writeThroughput.isEmpty) Some("25")
       else None
 
-    val renamedDf = renames
-      .foldLeft(df) {
-        case (acc, Rename(from, to)) => acc.withColumnRenamed(from, to)
+    val jobConf = new JobConf(spark.sparkContext.hadoopConfiguration)
+
+    def setOptionalConf(name: String, maybeValue: Option[String]): Unit =
+      for (value <- maybeValue) {
+        jobConf.set(name, value)
+      }
+
+    jobConf.set(DynamoDBConstants.OUTPUT_TABLE_NAME, target.table)
+    setOptionalConf(DynamoDBConstants.REGION, target.region)
+    setOptionalConf(DynamoDBConstants.ENDPOINT, target.endpoint.map(_.renderEndpoint))
+    setOptionalConf(DynamoDBConstants.READ_THROUGHPUT, throughput)
+    setOptionalConf(
+      DynamoDBConstants.THROUGHPUT_WRITE_PERCENT,
+      target.throughputWritePercent.map(_.toString))
+    setOptionalConf(DynamoDBConstants.SCAN_SEGMENTS, target.scanSegments.map(_.toString))
+    setOptionalConf(DynamoDBConstants.MAX_MAP_TASKS, target.maxMapTasks.map(_.toString))
+    setOptionalConf(DynamoDBConstants.DYNAMODB_ACCESS_KEY_CONF, target.credentials.map(_.accessKey))
+    setOptionalConf(DynamoDBConstants.DYNAMODB_SECRET_KEY_CONF, target.credentials.map(_.secretKey))
+    jobConf.set(
+      "mapred.output.format.class",
+      "org.apache.hadoop.dynamodb.write.DynamoDBOutputFormat")
+    jobConf.set("mapred.input.format.class", "org.apache.hadoop.dynamodb.read.DynamoDBInputFormat")
+
+    rdd
+      .mapValues { itemWritable =>
+        val item = itemWritable.getItem
+        for (rename <- renames) {
+          item.put(rename.to, item.get(rename.from))
+          item.remove(rename.from)
```

Comment:
ok, I guess this would work and it could be also properly split and won't have much overhead I suppose
wondering how cassandra DF handles it (maybe they have the same code in their connector)